### PR TITLE
Makes you deaf while unconscious

### PR DESCRIPTION
**Pull request by Maciekkub**
## About The Pull Request
When you are unconscious, you won't hear any sounds. When you exit the state, your hearing shall be back

## Why It's Good For The Game
Only natural for a person to not be able to hear while he's unconscious
## Changelog
:cl:
tweak: toggles deaf when unconscious
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/life.dm [hunk 1]
```diff
@@ -55,10 +55,12 @@
 	if(knocked_out || sleeping || getOxyLoss() > CARBON_KO_OXYLOSS || health < get_crit_threshold())
 		if(stat != UNCONSCIOUS)
 			blind_eyes(1)
+			set_ear_damage(deaf = 30000)
 		stat = UNCONSCIOUS
 	else if(stat == UNCONSCIOUS)
 		stat = CONSCIOUS
 		adjust_blindness(-1)
+		ear_deaf = 0
```
Comment (DominikPanic) [inline code/modules/mob/living/carbon/life.dm]: Don't modify values directly, always use the helpers.

File: code/modules/mob/living/carbon/life.dm [hunk 2]
```diff
@@ -55,10 +55,12 @@
 	if(knocked_out || sleeping || getOxyLoss() > CARBON_KO_OXYLOSS || health < get_crit_threshold())
 		if(stat != UNCONSCIOUS)
 			blind_eyes(1)
+			set_ear_damage(deaf = 30000)
```
Comment (DominikPanic) [inline code/modules/mob/living/carbon/life.dm]: This may be better handled via the deaf disability.

File: code/modules/mob/living/carbon/life.dm [hunk 3]
```diff
@@ -55,6 +55,7 @@
 	if(knocked_out || sleeping || getOxyLoss() > CARBON_KO_OXYLOSS || health < get_crit_threshold())
 		if(stat != UNCONSCIOUS)
 			blind_eyes(1)
+			DEAF(1<<2)
```
Comment (psykzz) [inline code/modules/mob/living/carbon/life.dm]:
```suggestion
			M.disabilities |= DEAF
```

`M.disabilities` is a bitfield, which can store multiple types of bit shifted values (things assigned like `1<<2`).
When you do `|=` This is like saying add `DEAF` to `M.disabilties`, but in a bitfield way.

File: code/modules/mob/living/carbon/life.dm [hunk 4]
```diff
@@ -55,6 +55,7 @@
...
 		stat = UNCONSCIOUS
 	else if(stat == UNCONSCIOUS)
 		stat = CONSCIOUS
```
Comment (psykzz) [inline code/modules/mob/living/carbon/life.dm]: Addtionally don't forget to remove the deafness when they are no longer unconscious

File: code/modules/mob/living/carbon/life.dm [hunk 5]
```diff
@@ -54,7 +54,9 @@
 
 	if(knocked_out || sleeping || getOxyLoss() > CARBON_KO_OXYLOSS || health < get_crit_threshold())
 		if(stat != UNCONSCIOUS)
+			var/mob/living/M
```
Comment (DominikPanic) [inline code/modules/mob/living/carbon/life.dm]: Remove this line, you're already in mob/living/carbon

File: code/modules/mob/living/carbon/life.dm [hunk 6]
```diff
@@ -54,7 +54,9 @@
...
 			blind_eyes(1)
+			stat = M.disabilities &= DEAF
```
Comment (DominikPanic) [inline code/modules/mob/living/carbon/life.dm]:
```suggestion
			stat = disabilities &= DEAF
```

File: code/modules/mob/living/carbon/life.dm (hunk 4 above)
Comment (DominikPanic) [reply code/modules/mob/living/carbon/life.dm]: Don't forget this

File: code/modules/mob/living/carbon/life.dm [hunk 7]
```diff
@@ -55,10 +55,12 @@
 	if(knocked_out || sleeping || getOxyLoss() > CARBON_KO_OXYLOSS || health < get_crit_threshold())
 		if(stat != UNCONSCIOUS)
 			blind_eyes(1)
+			stat = disabilities |= DEAF
 		stat = UNCONSCIOUS
 	else if(stat == UNCONSCIOUS)
 		stat = CONSCIOUS
 		adjust_blindness(-1)
+		disabilities = 0
```
Comment (DominikPanic) [inline code/modules/mob/living/carbon/life.dm]: This will remove all disabilities, not just deaf

File: code/modules/mob/living/carbon/life.dm [hunk 8]
```diff
@@ -55,10 +55,12 @@
 	if(knocked_out || sleeping || getOxyLoss() > CARBON_KO_OXYLOSS || health < get_crit_threshold())
 		if(stat != UNCONSCIOUS)
 			blind_eyes(1)
+			stat = disabilities |= DEAF
```
Comment (DominikPanic) [inline code/modules/mob/living/carbon/life.dm]:
```suggestion
			disabilities |= DEAF
```

**Pull request merged**